src/interceptors/oauth-interceptor.js

```diff
@@ -8,9 +8,10 @@
 function oauthInterceptor($q, $rootScope, OAuthToken) {
   return {
     request: function(config) {
+      config.headers = config.headers || {};
+
       // Inject `Authorization` header.
-      if (OAuthToken.getAuthorizationHeader()) {
-        config.headers = config.headers || {};
+      if (!config.headers.hasOwnProperty('Authorization') && OAuthToken.getAuthorizationHeader()) {
```

Comment:
I saw `lodash` as dependency on `package.json` and jumped too fast on this :disappointed_relieved: